src/vs/platform/terminal/common/capabilities/commandDetection/promptInputModel.ts

```diff
@@ -280,13 +280,17 @@ export class PromptInputModel extends Disposable implements IPromptInputModel {
 
 		const buffer = this._xterm.buffer.active;
 		let line = buffer.getLine(commandStartY);
+		const absoluteCursorY = buffer.baseY + buffer.cursorY;
+		let cursorIndex: number | undefined;
+
 		let commandLine = line?.translateToString(true, this._commandStartX);
-		if (!line || commandLine === undefined) {
+		if (!line || !commandLine) {
```

Comment:
I think we can only do this `commandLine === ''` check for fish, otherwise it's going to trace log `PromptInputModel#_sync: no line` for perfectly valid prompt inputs where there's no input yet.

Instead of combining it with this block, how about we pull it out of the `if (!line || commandLine === undefined) {` block and have another block that modifies commandLine:

```ts
if (this._shellType === PosixShellType.Fish) {
  // If commandLine is '' AND cursorY > commandStartY
    // If commandStartY has content
      // Block content...
```